net: trickle: Debug the trickle timer for multiple triggerings

lingwei901011 opened the pull request:
fixed issue #15606. for multiple triggerings, we need re-initialize
the delayed work "trickle->timer" to change its handler function to
trickle_timeout()

Signed-off-by: ling wei <lingwei@cisco.com>

jukkar commented on subsys/net/ip/trickle.c:
```diff
@@ -21,6 +21,9 @@ LOG_MODULE_REGISTER(net_trickle, CONFIG_NET_TRICKLE_LOG_LEVEL);
 
```
Please fix the commit message. The commit subject looks a bit weird, I do not get the "Debug" word here. Perhaps the commit subject could look like `net: trickle: Re-init trickle timer for multiple triggerings` or something like that.

Also the commit body could be more clearer, so something like
```
We need to re-initialize the trickle->timer delayed work to use
trickle_timeout() if there are multiple triggerings.

Fixes #15606 
```

jukkar commented on subsys/net/ip/trickle.c:
```diff
@@ -21,6 +21,9 @@ LOG_MODULE_REGISTER(net_trickle, CONFIG_NET_TRICKLE_LOG_LEVEL);
 
 #define TICK_MAX ~0
 
+/* function declaration in advance */
```
Remove this comment, no need to comment obvious things.

jukkar reviewed (changes requested):
Looks good, minor tweaks needed.

lingwei901011 committed:
net: trickle: Re-init trickle timer for multiple triggerings

We need to re-initialize the trickle->timer delayed work to use
trickle_timeout() if there are multiple triggerings.

Fixes #15606

Signed-off-by: ling wei <lingwei@cisco.com>
